IdentityCore/src/throttling/model/MSIDThrottlingModelBase.h

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+#import "MSIDThumbprintCalculatable.h"
+#import "MSIDThrottlingCacheRecord.h"
+#import "NSError+MSIDThrottlingExtension.h"
+#import "MSIDRequestContext.h"
+#import "MSIDLRUCache.h"
+typedef NS_ENUM(NSInteger, MSIDThrottlingType)
+{
+    MSIDThrottlingTypeNone = 0,
+    MSIDThrottlingType429 = 1,
+    MSIDThrottlingTypeInteractiveRequired = 2
+};
+
+typedef NS_ENUM(NSInteger, MSIDThrottlingThumbprintType)
+{
+    MSIDThrottlingThumbprintTypeStrict = 0,
+    MSIDThrottlingThumbprintTypeFull = 1
+};
+
+NS_ASSUME_NONNULL_BEGIN
+@interface MSIDThrottlingModelBase : NSObject
+
+@property NSString *thumbprintValue;
```

Comment:
nit: do we want to make some of them readonly & did you want to include nonatomic flags as well? 
(nonatomic, readonly)

Reply:
and then we create getters in the derived classes' respective implementation files. I think that's the recommended practice. @oldalton

Reply:
Agree with the readonly & non atomic, I updated some properties on that.
I don't understand about the getters in the derived classes yet, can you give me an example in any these fields above?

Reply:
We create readonly properties in .h file, and in .m file you create a function that returns the same object type and has the same name as the property.

Reply:
Oh ok, but I thought we want to do that if we need to do something before return the value. In this case, I don't do anything other than assigning that property a value in the initializer and then later retrieve the value, do I still need that setup?